sccp/figures/common.py

```diff
@@ -142,12 +143,23 @@ def plotFactors(factors, data: Pf2X, axs, reorder=tuple(), trim=tuple(), saveGen
                 cmap=cmap,
             )
 
-        if i == 0 and not (row_colors is None):
+        if i == 0 and not (cond_group_labels is None):
             # add little boxes to denote SLE/healthy rows
             axs[i].tick_params(axis='y', which='major', pad=20, length=0) # extra padding to leave room for the row colors
+            # get list of colors for each label:
+            colors = sns.color_palette(n_colors = pd.Series(cond_group_labels).nunique()).as_hex()
+            lut = {}
+            legend_elements = []
+            for index, group in enumerate(pd.Series(cond_group_labels).unique()):
+                lut[group] = colors[index]
+                legend_elements.append(Patch(color = colors[index],
+                                             label = group))
+            row_colors = pd.Series(cond_group_labels).map(lut)
             for iii, color in enumerate(row_colors):
                 axs[i].add_patch(plt.Rectangle(xy=(-0.05, iii), width=0.05, height=1, color=color, lw=0,
-                             transform=axs[i].get_yaxis_transform(), clip_on=False))
+                                transform=axs[i].get_yaxis_transform(), clip_on=False))
+            # add a little legend
+            axs[i].legend(handles = legend_elements, bbox_to_anchor = (0.18, 1.07))
 
```

Comment:
Before, `row_colors` had to be defined as a mapping between the column of interest (`SLE_status`) and the color you wanted (see lines 46 and 47 that get REMOVED from `figureS1.py`). Now, the function just takes in an array of values in the order of the samples, rearranges those if the samples get rearranged, and then calculates how many colors it needs and makes a legend for those colors